src/cli/search.rs

```diff
@@ -0,0 +1,160 @@
+use std::{cmp::Ordering, path::PathBuf};
+
+use clap::Parser;
+use itertools::Itertools;
+use miette::IntoDiagnostic;
+use rattler_conda_types::{Channel, ChannelConfig, Platform, RepoDataRecord};
+use rattler_repodata_gateway::sparse::SparseRepoData;
+use strsim::jaro;
+use tokio::task::spawn_blocking;
+
+use crate::{progress::await_in_progress, repodata::fetch_sparse_repodata, Project};
+
+/// Search a package, output will list the latest version of package
+#[derive(Debug, Parser)]
+#[clap(arg_required_else_help = true)]
+pub struct Args {
+    /// Name of package to search
+    #[arg(required = true)]
+    pub package: String,
+
+    /// Channel to specifically search package
+    #[clap(short, long, default_values = ["conda-forge"])]
+    channel: Vec<String>,
+
+    /// The path to 'pixi.toml'
+    #[arg(long)]
+    pub manifest_path: Option<PathBuf>,
+
+    /// Limit the number of search results
+    #[clap(short, long, default_value_t = 15)]
+    limit: usize,
+}
+
+/// fetch packages from `repo_data` based on `filter_func`
+fn search_package_by_filter<F>(
+    package: &str,
+    repo_data: &[SparseRepoData],
+    filter_func: F,
+) -> miette::Result<Vec<RepoDataRecord>>
+where
+    F: Fn(&str, &str) -> bool,
+{
+    let similar_packages = repo_data
+        .iter()
+        .flat_map(|repo| {
+            repo.package_names()
+                .filter(|&name| filter_func(name, package))
+        })
+        .collect::<Vec<&str>>();
+
+    let mut latest_packages = Vec::new();
+
+    // search for `similar_packages` in all platform's repodata
+    // add the latest version of the fetched package to latest_packages vector
+    for repo in repo_data {
+        for package in &similar_packages {
+            let mut records = repo.load_records(package).into_diagnostic()?;
+            // sort records by version, get the latest one
+            records.sort_by(|a, b| a.package_record.version.cmp(&b.package_record.version));
+            let latest_package = records.last().cloned();
+            if let Some(latest_package) = latest_package {
+                latest_packages.push(latest_package);
+            }
+        }
+    }
+
+    latest_packages = latest_packages
+        .into_iter()
+        .unique_by(|record| record.package_record.name.clone())
+        .collect::<Vec<_>>();
+
+    Ok(latest_packages)
+}
+
+pub async fn execute(args: Args) -> miette::Result<()> {
+    let project = Project::load_or_else_discover(args.manifest_path.as_deref()).ok();
+
+    let channels = if let Some(project) = project {
```

Comment:
This might give some unexpected behavior. As I would expect defining channels should overwrite the things it defaults to. 
So if there is something else then `conda-forge` defined it should ignore the project. And act like being outside of the project.

Reply:
if it's being ignored in case it's not conda-forge, is there even any need for it?

Reply:
Ah yeah, 

let me try to give an example:
```
~/pixi_project $ pixi search hello
# Uses the channels defined in the pixi_project/pixi.toml

~ $ pixi search hello
# Searches in conda-forge

~ $ pixi search --channel bioconda hello
# Only searches in bioconda

~/pixi_project $ pixi search --bioconda hello
# Should also only search in bioconda
# This case is ignored in your implementation and it will search in the channels defined in the project.
```

Reply:
Oh yes, this definitely makes more sense. Will change this ASAP